src/jdk.jpackage/share/classes/jdk/jpackage/internal/IOUtils.java

```diff
@@ -220,4 +224,33 @@ public static long getPID(Process p) {
             return -1;
         }
     }
+
+    static Map<String, String> getPropertiesFromFile(Path filename) {
+        Map<String, String> map = new HashMap<>();
+        // load properties file
+        Properties properties = new Properties();
+        try (Reader reader = Files.newBufferedReader(filename)) {
+            properties.load(reader);
+        } catch (IOException e) {
+            Log.error("Exception: " + e.getMessage());
+        }
+
+        for (final String name: properties.stringPropertyNames()) {
+            map.put(name, properties.getProperty(name));
+        }
+
+        return map;
+    }
+
+    public static String getPropertyFromFile(Path filename, String name) {
+        // load properties file
```

Comment:
That is a bad idea to expose jpackage API that uses jpackage's Log class. Where will these log messages go?

Reply:
Fixed.